components/org.wso2.carbon.identity.recovery/src/main/java/org/wso2/carbon/identity/recovery/handler/UserSelfRegistrationHandler.java

```diff
@@ -122,6 +122,33 @@ public void handleEvent(Event event) throws IdentityEventException {
         if (IdentityEventConstants.Event.POST_ADD_USER.equals(event.getEventName())) {
             UserRecoveryDataStore userRecoveryDataStore = JDBCRecoveryDataStore.getInstance();
 
+            if (isAccountLockOnCreation || isEnableConfirmationOnCreation) {
+                HashMap<String, String> userClaims = new HashMap<>();
+                if (isAccountLockOnCreation) {
+                    // Need to lock user account.
+                    userClaims.put(IdentityRecoveryConstants.ACCOUNT_LOCKED_CLAIM, Boolean.TRUE.toString());
+                    userClaims.put(IdentityRecoveryConstants.ACCOUNT_LOCKED_REASON_CLAIM,
+                            IdentityMgtConstants.LockedReason.PENDING_SELF_REGISTRATION.toString());
+                }
+                if (Utils.isAccountStateClaimExisting(tenantDomain)) {
+                    userClaims.put(IdentityRecoveryConstants.ACCOUNT_STATE_CLAIM_URI,
+                            IdentityRecoveryConstants.PENDING_SELF_REGISTRATION);
+                }
+                try {
+                    userStoreManager.setUserClaimValues(user.getUserName(), userClaims, null);
+                    if (log.isDebugEnabled()) {
+                        if (isAccountLockOnCreation) {
+                            log.debug("Locked user account: " + user.getUserName());
+                        }
+                        if (isEnableConfirmationOnCreation) {
+                            log.debug("Send verification notification for user account: " + user.getUserName());
+                        }
+                    }
+                } catch (UserStoreException e) {
+                    throw new IdentityEventException("Error while lock user account :" + user.getUserName(), e);
+                }
+            }
+
```

Comment:
There is a return from here https://github.com/wso2-extensions/identity-governance/pull/895/files#diff-8a0d606a915a355053e3ac3ab16564269874a60d46fa5bf2cee7591fceca8658R162. We may have to consider that logic before locking the account. Only the notification sending part is required to be moved down.